kwonhayeon/SpringPrac/src/main/java/com/likelion/springprac/repository/UserRepository.java

```diff
@@ -0,0 +1,57 @@
+package com.likelion.springprac.repository;
+
+import com.likelion.springprac.domain.User;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+// Spring이 이 클래스를 저장소(데이터 관리) 역할로 인식하게 함
+public class UserRepository {
+
+    //user 데이터를 임시 데이터베이스처럼 사용하도록 저장하는 리스트
+    private final List<User> userList = new ArrayList<>();
+    // 사용자 ID를 자동으로 증가시키기 위한 변수
+    private Long idCounter = 1L;
+
+    // Create (데이터 생성)
+    // 새로운 user를 리스트에 저장하고, ID를 자동 부여한 후 반환
+    public User save(User user) {
+        user.setId(idCounter++); // ID 자동 증가
+        userList.add(user); // 리스트에 user 추가
+        return user; // 저장된 user 반환
+    }
+
+    // Read (모든 user 조회)
+    // 현재 저장된 전체 사용자 목록을 반환함
+    public List<User> findAll() {
+        return userList;
```

Comment:
userList를 직접 반환하게 되면 외부에서 리스트를 수정할 수 있는 위험이 있어요!
불변리스트로 감싸는 방식을 사용하면 더 안전한 코드가 될 수 있을 것 같아요 😊😊

```suggestion
        return List.copyOf(userList);
```